sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -294,8 +294,8 @@ query
     ;
 
 insertInto
-    : INSERT OVERWRITE TABLE tableIdentifier (partitionSpec (IF NOT EXISTS)?)?                              #insertOverwriteTable
-    | INSERT INTO TABLE? tableIdentifier partitionSpec?                                                     #insertIntoTable
+    : INSERT OVERWRITE TABLE? multipartIdentifier (partitionSpec (IF NOT EXISTS)?)?                         #insertOverwriteTable
+    | INSERT INTO TABLE? multipartIdentifier partitionSpec? (IF NOT EXISTS)?                                #insertIntoTable
```

Comment:
I think `(partitionSpec (IF NOT EXISTS)?)?` is better? AFAIK there is such an operation in SQL standard that can insert into a table if the table exists. The `IF NOT EXISTS` is only for partitions.

Reply:
Actually, `IF NOT EXISTS` doesn't make sense to partition either. It's append not overwrite, and it seems weird to me if we can't append to an existing partition.

Reply:
Can we keep this unchanged and not add `IF NOT EXISTS` here?

Reply:
This was changed to get a better error message. Instead of a parse exception that lists symbols, this is now a useful error message with a test.